website/docs/d/datasource_google_service_account_id_token.html.markdown

```diff
@@ -0,0 +1,105 @@
+---
+subcategory: "Cloud Platform"
+layout: "google"
+page_title: "Google: google_service_account_id_token"
+sidebar_current: "docs-google-service-account-id-token"
+description: |-
+  Produces OpenID Connect token for service accounts
+---
+
+# google\_service\_account\id\_token
+
+This data source provides a google OpenID Connect (`oidc`) `id_token`.  Tokens issued using this datasource are typically used to call external services that accept OIDC tokens for authentication (e.g. [Google Cloud Run](https://cloud.google.com/run/docs/authenticating/service-to-service)). 
+
+For more information see
+[OpenID Connect](https://openid.net/specs/openid-connect-core-1_0.html#IDToken).
+
+## Example Usage - ServiceAccount JSON credential file.
+  Datasource will the service account provided by `GOOGLE_APPLICATION_CREDENTIALS` or `GOOGLE_CLOUD_KEYFILE_JSON` environment variable. 
+  
+  ```hcl
+  data "google_service_account_id_token" oidc {
+    target_audience = "https://foo.bar/"
+  }
+
+  output "oidc_token" {
+    value = data.google_service_account_id_token.oidc.id_token
+  }
+  ```
+
+## Example Usage - `Application Default Credentials`
+  Datasource will automatically use credentials provided by [Application Default Credentials](https://cloud.google.com/docs/authentication/production)
+
+## Example Usage - Service Account Impersonation.
+  Datasource will use background impersonated credentials provided by [google_service_account_access_token](https://www.terraform.io/docs/providers/google/d/datasource_google_service_account_access_token.html).  
+
+  Note: to use the following, you must grant `target_service_account` the
```

Comment:
Does it make sense to have this as an iam_member resource in the example?

Reply:
i think that part is included in the docs about impersonation datasource though. ...it'll make it a bit long here to include the full flow.  IMO, the impersonation thing isn't that common vs plain svc account id token stuff